monadic-rx/shared/src/main/scala/mhtml/rx.scala

```diff
@@ -250,6 +264,22 @@ object Rx {
         Cancelable { () => cc.cancel; self.imitating = false }
       } else run(other)(effect)
 
+    case shareRx @ Sharing(self) =>
```

Comment:
It's a small detail but I think that when using `@` like this there is a conversion to shadow the original name such to give it a more precise type:

```scala
rx match {
   ...
   case rx @ Sharing(self) =>
}
```